app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java

```diff
@@ -302,16 +338,77 @@ private void loadSubscriptions() {
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(
-                    result -> {
-                        listItems = result;
-                        subscriptionAdapter.setItems(result);
-                        subscriptionAdapter.notifyDataSetChanged();
-                        emptyView.updateVisibility();
-                        progressBar.setVisibility(View.GONE); // Keep hidden to avoid flickering while refreshing
-                    }, error -> {
-                        Log.e(TAG, Log.getStackTraceString(error));
-                        progressBar.setVisibility(View.GONE);
-                    });
+                        result -> {
+                            tagFilteredFeedItems = new ArrayList<>();
+                            List<NavDrawerData.FolderDrawerItem> feedFolders = new ArrayList<>();
+                            for (NavDrawerData.DrawerItem drawerItem : result) {
+                                if (drawerItem.type.equals(NavDrawerData.DrawerItem.Type.FOLDER)) {
+                                    feedFolders.add((NavDrawerData.FolderDrawerItem) drawerItem);
+                                    if (((NavDrawerData.FolderDrawerItem) drawerItem).name.equals(FeedPreferences.TAG_ROOT)) {
+                                        rootFolder = (NavDrawerData.FolderDrawerItem) drawerItem;
+                                    }
+                                } else {
+                                    tagFilteredFeedItems.add(drawerItem);
+                                }
+                            }
+                            feedTagAdapter = new FeedTagAdapter(new ArrayList<>());
+                            Set<String> tagFilterIds = UserPreferences.getTagFilterIds();
+
+                            for (NavDrawerData.FolderDrawerItem folder : feedFolders) {
+                                if (tagFilterIds.contains(String.valueOf(folder.id))) {
+                                    feedTagAdapter.addItem(folder);
+                                }
+                            }
+                            Chip rootChip = null;
+                            folderChipGroup.removeAllViews();
+                            for (NavDrawerData.FolderDrawerItem folderItem : feedFolders) {
+                                Chip folderChip = new Chip(getActivity());
```

Comment:
Don't you already have an adapter for this? Also, this is too much code in the RxJava handler. Please extract to a method.

Reply:
No this is not a RecyclerView. It is ChipGroup, used for the second tag list. I don't think RecyclerView would work, since it doesn't provide wrapping (LinearLayout) or varying sizes for items (GridLayout)